src/message_pool.zig

```diff
@@ -65,6 +66,34 @@ comptime {
 /// initialization and reused thereafter. The messages_max values determine the size of this pool.
 pub const MessagePool = struct {
     pub const Message = struct {
+        pub const Reserved = MessageType(.reserved);
```

Comment:
Hm, it seems that `Message` and  `Message.Prepare` differ in the level of indirection: `Message.Type` is a pointer to `Message`. This is not idea, I think:

* first, because there's a difference between a raw and a type message
* second, because it adds one more level of indirection. Eg, in aof, when reading
  
  ```
  pub fn to_message(self: *AOFEntry, target: Message.Prepare) void {
  ```

  there's now extra effort reason "why, why we are passing a message by value? Ahh, that's because a message is secretly a pointer!"

Reply:
:thinking: The `MessageType(command)` types could cast the `Message` memory (instead of holding a pointer). Then the `MessageType(command)`s would be pointers too.

Thoughts? Fwiw I didn't go that route originally since it seems like extra "magic". Plus `Message`s aren't `extern struct`, so I don't think their layouts are actually guaranteed to be equivalent. (Though in practice I assume they would be.)

Reply:
> Then the MessageType(command)s would be pointers too.

Yeah, I think we should do this: I don't like "hiding" pointers in structs, by-value call site just looks too weird. 

Good point about extern structs! I think we could make message an extern one?